File: src/typescript/config/init-general/chain-with-config.ts
import { createHiveChain, IWaxOptionsChain } from '@hiveio/wax';

// Define custom options
const customOptions: IWaxOptionsChain = {
  // Example custom chain ID:
  chainId: 'f875a0b000000000000000000000000000000000000000000000000000000000',
  // Example custom API endpoint:
  apiEndpoint: 'https://hive.custom.endpoint',
  // Example custom REST API endpoint:
  restApiEndpoint: 'https://rest.api.custom.endpoint',
  // Disable API timeout:
  apiTimeout: 0
};

// Initialize Hive Chain with custom options
const chain = await createHiveChain(customOptions);

File: src/rust/config/init_general/chain_with_config.rs
use wax::prelude::*;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Define custom options
    let custom_options = HiveChainOptions {
        // Example custom chain ID:
        chain_id: "f875a0b000000000000000000000000000000000000000000000000000000000"
            .into(),
        // Example custom API endpoint:
        api_endpoint: "https://hive.custom.endpoint".into(),
        // Example custom REST API endpoint:
        rest_api_endpoint: "https://rest.api.custom.endpoint".into(),
        // Disable API timeout:
        api_timeout: 0,
        // Keep the defaults for everything else:
        ..Default::default()
    };

    // Initialize Hive Chain with custom options
    let _chain = create_hive_chain(custom_options)?;

    Ok(())
}
